### PR TITLE
Teachable Machine Learning Setup

### PR DESCRIPTION
**Pull request by jfewii**

### CODE REVIEW

File: portfolio/src/main/webapp/learn.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// More API functions here:
+// https://github.com/googlecreativelab/teachablemachine-community/tree/master/libraries/image
+
+// the link to your model provided by Teachable Machine export panel
+
```
Comment (jgao23) [inline portfolio/src/main/webapp/learn.js:5]: comments should be right above the code, no need for new empty line

File: portfolio/src/main/webapp/learn.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+let model, webcam, labelContainer, maxPredictions;
+
+// Load the image model and setup the webcam
+async function init() {
+  const modelURL = URL + "model.json";
+  const metadataURL = URL + "metadata.json";
+
+  // load the model and metadata
+  // Refer to tmImage.loadFromFiles() in the API to support files from a file picker
+  // or files from your local hard drive
```
Comment (jgao23) [inline portfolio/src/main/webapp/learn.js:17]: was this copied from somewhere? if you're not using loadFromFiles() no need to reference it

File: portfolio/src/main/webapp/learn.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+// More API functions here:
```
Comment (jgao23) [inline portfolio/src/main/webapp/learn.js:1]: Can you add some more comments hhere about what you expect this learn.js file to do?

File: portfolio/src/main/webapp/learn.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  await webcam.setup(); // request access to the webcam
+  await webcam.play();
+  window.requestAnimationFrame(loop);
+
+  // append elements to the DOM
+  document.getElementById("webcam-container").appendChild(webcam.canvas);
+  labelContainer = document.getElementById("label-container");
+  for (let i = 0; i < maxPredictions; i++) { // and class labels
+    labelContainer.appendChild(document.createElement("div"));
+  }
```
Comment (jgao23) [inline portfolio/src/main/webapp/learn.js:35]: you're initializing the DOM here right? why do you need to initialize it here, can you not create each element in predict()?

File: portfolio/src/main/webapp/learn.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  // Refer to tmImage.loadFromFiles() in the API to support files from a file picker
+  // or files from your local hard drive
+  // Note: the pose library adds "tmImage" object to your window (window.tmImage)
+
+  model = await tmImage.load(modelURL, metadataURL);
+  maxPredictions = model.getTotalClasses();
+
+  // Convenience function to setup a webcam
+  const flip = true; // whether to flip the webcam
+  webcam = new tmImage.Webcam(200, 200, flip); // width, height, flip
```
Comment (andreahu12) [inline portfolio/src/main/webapp/learn.js:20]: i know this was copied form the sample code, but let's make it more readable!

` webcam = new tmImage.Webcam(200, 200, flip); // width, height, flip`
 --> ` webcam = new tmImage.Webcam(/* width= */ 200, /* height= */ 200, flip);`

https://google.github.io/styleguide/jsguide.html#formatting-param-name-comments

**Comment by andreahu12** [comment]
Also could rename PR title to be more descriptive. Typically should start with a verb like "Use teachable machine to classify webcam images."

File: portfolio/src/main/webapp/learn.js (hunk 3 above)
Comment (jgao23) [reply portfolio/src/main/webapp/learn.js:2]: i don't think you added comments here

**Pull request merged**